### PR TITLE
add da1469x SPI bus driver

### PR DESCRIPTION
**Pull request by kasjer**
This adds SPI bus driver that utilizes DMA

MCU da1469x_periph_create_spi() now creates new SPI bus(es) using new driver.

### CODE REVIEW

**Review by wes3 — COMMENTED** [review]
Made some minor comments that do not require any changes; up to you if you want to address them or not.

File: hw/bus/drivers/spi_da1469x/src/spi_da1469x.c [hunk 1]
```diff
@@ -0,0 +1,665 @@
...
+end:
+    return rc;
+}
+
+static int
+spi_da1469x_read(struct bus_dev *bdev, struct bus_node *bnode,
+                 uint8_t *buf, uint16_t length, os_time_t timeout,
+                 uint16_t flags)
+{
+    static uint16_t mosi_data = 0xFFFF;
```
[the remark this answers is not in the document]
Comment (kasjer) [reply hw/bus/drivers/spi_da1469x/src/spi_da1469x.c]: No special reason to have 0xFFFF. Now it is changed to 0.
Both variable for dummy DMA are now similar.

File: hw/bus/drivers/spi_da1469x/src/spi_da1469x.c [hunk 2]
```diff
@@ -0,0 +1,665 @@
...
+static int
+spi_da1469x_configure(struct bus_dev *bdev, struct bus_node *bnode)
+{
+    struct bus_spi_dev *dev = (struct bus_spi_dev *)bdev;
+    struct bus_spi_node *node = (struct bus_spi_node *)bnode;
+    struct bus_spi_node *current_node = (struct bus_spi_node *)bdev->configured_for;
+    struct spi_da1469x_driver_data *dd;
+    SPI_Type *regs;
+    uint32_t ctrl_reg;
+    int rc = 0;
```
[the remark this answers is not in the document]
Comment (kasjer) [reply hw/bus/drivers/spi_da1469x/src/spi_da1469x.c]: yeah, why not

**Review by wes3 — APPROVED** [review]
LGTM

**Pull request merged**